### PR TITLE
Update DotNet_Security_Cheat_Sheet.md

### PR DESCRIPTION
**Pull request by fcerullo-cycubix**
addition to A08 based on Microsoft documentation

Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [ ] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [ ] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [ ] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [ ] All your assets are stored in the **assets** folder.
- [ ] All the images used are in the **PNG** format.
- [ ] Any references to websites have been formatted as `[TEXT](URL)`
- [ ] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [ ] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #<insert number here>.

Thank you again for your contribution :smiley:

### CODE REVIEW

**Review by jmanico — DISMISSED** [review]

**Comment by jmanico** [comment]
Nice edit thank you! I am seeing some lint errors, can you kindly fix this?

cheatsheets/DotNet_Security_Cheat_Sheet.md:799 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- XmlSerializer and DataContra..."]

File: cheatsheets/DotNet_Security_Cheat_Sheet.md [hunk 1]
```diff
@@ -794,6 +794,12 @@ DO NOT: Send unsigned or unencrypted serialized objects over the network
 DO: Perform integrity checks or validate digital signatures on serialized
 objects received from the network
 
+DO NOT: Use the BinaryFormatter type which is dangerous and not recommended for data processing.
+NET offers several in-box serializers that can handle untrusted data safely:
```
Comment (szh) [inline cheatsheets/DotNet_Security_Cheat_Sheet.md:798]:
```suggestion
.NET offers several in-box serializers that can handle untrusted data safely:
```

File: cheatsheets/DotNet_Security_Cheat_Sheet.md [hunk 2]
```diff
@@ -794,6 +794,12 @@ DO NOT: Send unsigned or unencrypted serialized objects over the network
 DO: Perform integrity checks or validate digital signatures on serialized
 objects received from the network
 
+DO NOT: Use the BinaryFormatter type which is dangerous and not recommended for data processing.
```
Comment (szh) [inline cheatsheets/DotNet_Security_Cheat_Sheet.md:797]:
```suggestion
DO NOT: Use the BinaryFormatter type which is dangerous and [not recommended](https://learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-security-guide) for data processing.
```

**Review by szh — CHANGES_REQUESTED** [review]
Looks good! Just added a missing `.` for `.NET` and a link to the MS docs

**Comment by wittjoe1** [comment]
@fcerullo-cycubix:

Can you please check whether some of the content is now obsolete?

The chapter on .net refers to version 4.5 - nobody should really be using that anymore. It contains references to dotnet_security_unit_testing - this project was created over 7 years ago. Is this still up to date or would it not be better to exclude it in order to avoid a false sense of security?

**Review by jmanico — DISMISSED** [review]

**Comment by jmanico** [comment]
I still see one lint error:

cheatsheets/DotNet_Security_Cheat_Sheet.md:799 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- XmlSerializer and DataContra..."]

**Comment by jmanico** [comment]
Lint fixes and @szh 's commits all integrated and merged!

**Pull request merged**